Tests fail, seems like Cython is not configured to compile with numpy correctly

Following are the standard output:

	D:\sympytest>git clone git@github.com:sympy/sympy.git
	Cloning into 'sympy'...
	remote: Counting objects: 153742, done.
	remote: Compressing objects: 100% (4/4), done.
	Receiving objects: 100% (153742/153742), 62.98 MiB | 13.00 KiB/s, done.

	Resolving deltas: 100% (122073/122073), done.
	Checking connectivity... done

	sh-3.1$ cd sympy

	sh-3.1$ ./bin/test |tee test.log
	======================================================= test process starts =======================================================
	executable:         c:\Python\python.exe  (2.7.8-final-0) [CPython]
	architecture:       32-bit
	cache:              yes
	ground types:       python
	random seed:        68232668
	hash randomization: on (PYTHONHASHSEED=2780600447)

	sympy\assumptions\tests\test_assumptions_2.py[5] .....                                                                         [OK]
	sympy\assumptions\tests\test_context.py[4] ....                                                                                [OK]
	sympy\assumptions\tests\test_matrices.py[19] ..........ff.......                                                               [OK]



	__________________________________________________________ xpassed tests __________________________________________________________
	sympy\core\tests\test_wester.py: test_V12
	sympy\core\tests\test_wester.py: test_T10
	sympy\printing\tests\test_gtk.py: test_1
	sympy\utilities\tests\test_module_imports.py: test_module_imports_are_direct

	___________________________________________________________________________________________________________________________________
	_________________________________ sympy\external\tests\test_autowrap.py:test_wrap_twice_c_cython __________________________________
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 138, in test_wrap_twice_c_cython
	    runtest_autowrap_twice('C', 'cython')
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 51, in runtest_autowrap_twice
	    f = autowrap((((a + b)/c)**5).expand(), language, backend)
	  File "sympy\core\cache.py", line 91, in wrapper
	    retval = cfunc(*args, **kwargs)
	  File "sympy\core\compatibility.py", line 872, in wrapper
	    result = user_function(*args, **kwds)
	  File "sympy\utilities\autowrap.py", line 508, in autowrap
	    return code_wrapper.wrap_code(routine, helpers=helps)
	  File "sympy\utilities\autowrap.py", line 151, in wrap_code
	    shutil.rmtree(workdir)
	  File "c:\Python\lib\shutil.py", line 252, in rmtree
	    onerror(os.remove, fullname, sys.exc_info())
	  File "c:\Python\lib\shutil.py", line 250, in rmtree
	    os.remove(fullname)
	WindowsError: [Error 5] Access is denied: 'c:\\users\\pecker\\appdata\\local\\temp\\tmpgz2ecp_sympy_compile\\wrapper_module_1.pyd'
	___________________________________________________________________________________________________________________________________
	_______________________________ sympy\external\tests\test_autowrap.py:test_autowrap_trace_C_Cython ________________________________
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 143, in test_autowrap_trace_C_Cython
	    runtest_autowrap_trace('C', 'cython')
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 61, in runtest_autowrap_trace
	    trace = autowrap(A[i, i], language, backend)
	  File "sympy\core\cache.py", line 91, in wrapper
	    retval = cfunc(*args, **kwargs)
	  File "sympy\core\compatibility.py", line 872, in wrapper
	    result = user_function(*args, **kwds)
	  File "sympy\utilities\autowrap.py", line 508, in autowrap
	    return code_wrapper.wrap_code(routine, helpers=helps)
	  File "sympy\utilities\autowrap.py", line 144, in wrap_code
	    self._process_files(routine)
	  File "sympy\utilities\autowrap.py", line 163, in _process_files
	    " ".join(command), e.output.decode()))
	CodeWrapError: Error while executing command: c:\Python\python.exe setup.py build_ext --inplace. Command output is:
	running build_ext
	cythoning wrapper_module_2.pyx to wrapper_module_2.c
	building 'wrapper_module_2' extension
	creating build
	creating build\temp.win-amd64-2.7
	creating build\temp.win-amd64-2.7\Release
	c:\Python\MinGW\bin\gcc.exe -DMS_WIN64 -mdll -O -Wall -Ic:\Python\include -Ic:\Python\PC -c wrapper_module_2.c -o build\temp.win-amd64-2.7\Release\wrapper_module_2.o -std=c99
	wrapper_module_2.c:232:31: fatal error: numpy/arrayobject.h: No such file or directory
	compilation terminated.
	error: command 'c:\\Python\\MinGW\\bin\\gcc.exe' failed with exit status 1

	___________________________________________________________________________________________________________________________________
	___________________________ sympy\external\tests\test_autowrap.py:test_autowrap_matrix_vector_C_cython ____________________________
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 148, in test_autowrap_matrix_vector_C_cython
	    runtest_autowrap_matrix_vector('C', 'cython')
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 69, in runtest_autowrap_matrix_vector
	    mv = autowrap(expr, language, backend)
	  File "sympy\core\cache.py", line 91, in wrapper
	    retval = cfunc(*args, **kwargs)
	  File "sympy\core\compatibility.py", line 872, in wrapper
	    result = user_function(*args, **kwds)
	  File "sympy\utilities\autowrap.py", line 508, in autowrap
	    return code_wrapper.wrap_code(routine, helpers=helps)
	  File "sympy\utilities\autowrap.py", line 144, in wrap_code
	    self._process_files(routine)
	  File "sympy\utilities\autowrap.py", line 163, in _process_files
	    " ".join(command), e.output.decode()))
	CodeWrapError: Error while executing command: c:\Python\python.exe setup.py build_ext --inplace. Command output is:
	running build_ext
	cythoning wrapper_module_3.pyx to wrapper_module_3.c
	building 'wrapper_module_3' extension
	creating build
	creating build\temp.win-amd64-2.7
	creating build\temp.win-amd64-2.7\Release
	c:\Python\MinGW\bin\gcc.exe -DMS_WIN64 -mdll -O -Wall -Ic:\Python\include -Ic:\Python\PC -c wrapper_module_3.c -o build\temp.win-amd64-2.7\Release\wrapper_module_3.o -std=c99
	wrapper_module_3.c:232:31: fatal error: numpy/arrayobject.h: No such file or directory
	compilation terminated.
	error: command 'c:\\Python\\MinGW\\bin\\gcc.exe' failed with exit status 1

	___________________________________________________________________________________________________________________________________
	___________________________ sympy\external\tests\test_autowrap.py:test_autowrap_matrix_matrix_C_cython ____________________________
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 153, in test_autowrap_matrix_matrix_C_cython
	    runtest_autowrap_matrix_matrix('C', 'cython')
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 81, in runtest_autowrap_matrix_matrix
	    matmat = autowrap(expr, language, backend)
	  File "sympy\core\cache.py", line 91, in wrapper
	    retval = cfunc(*args, **kwargs)
	  File "sympy\core\compatibility.py", line 872, in wrapper
	    result = user_function(*args, **kwds)
	  File "sympy\utilities\autowrap.py", line 508, in autowrap
	    return code_wrapper.wrap_code(routine, helpers=helps)
	  File "sympy\utilities\autowrap.py", line 144, in wrap_code
	    self._process_files(routine)
	  File "sympy\utilities\autowrap.py", line 163, in _process_files
	    " ".join(command), e.output.decode()))
	CodeWrapError: Error while executing command: c:\Python\python.exe setup.py build_ext --inplace. Command output is:
	running build_ext
	cythoning wrapper_module_4.pyx to wrapper_module_4.c
	building 'wrapper_module_4' extension
	creating build
	creating build\temp.win-amd64-2.7
	creating build\temp.win-amd64-2.7\Release
	c:\Python\MinGW\bin\gcc.exe -DMS_WIN64 -mdll -O -Wall -Ic:\Python\include -Ic:\Python\PC -c wrapper_module_4.c -o build\temp.win-amd64-2.7\Release\wrapper_module_4.o -std=c99
	wrapper_module_4.c:232:31: fatal error: numpy/arrayobject.h: No such file or directory
	compilation terminated.
	error: command 'c:\\Python\\MinGW\\bin\\gcc.exe' failed with exit status 1

	___________________________________________________________________________________________________________________________________
	__________________________________ sympy\external\tests\test_autowrap.py:test_ufuncify_C_Cython ___________________________________
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 158, in test_ufuncify_C_Cython
	    runtest_ufuncify('C', 'cython')
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 93, in runtest_ufuncify
	    fabc = ufuncify([a, b, c], a*b + c, backend=backend)
	  File "sympy\core\cache.py", line 91, in wrapper
	    retval = cfunc(*args, **kwargs)
	  File "sympy\core\compatibility.py", line 857, in wrapper
	    return user_function(*args, **kwds)
	  File "sympy\utilities\autowrap.py", line 878, in ufuncify
	    tempdir, args, flags, verbose, helpers)
	  File "sympy\core\cache.py", line 91, in wrapper
	    retval = cfunc(*args, **kwargs)
	  File "sympy\core\compatibility.py", line 857, in wrapper
	    return user_function(*args, **kwds)
	  File "sympy\utilities\autowrap.py", line 508, in autowrap
	    return code_wrapper.wrap_code(routine, helpers=helps)
	  File "sympy\utilities\autowrap.py", line 144, in wrap_code
	    self._process_files(routine)
	  File "sympy\utilities\autowrap.py", line 163, in _process_files
	    " ".join(command), e.output.decode()))
	CodeWrapError: Error while executing command: c:\Python\python.exe setup.py build_ext --inplace. Command output is:
	running build_ext
	cythoning wrapper_module_5.pyx to wrapper_module_5.c
	building 'wrapper_module_5' extension
	creating build
	creating build\temp.win-amd64-2.7
	creating build\temp.win-amd64-2.7\Release
	c:\Python\MinGW\bin\gcc.exe -DMS_WIN64 -mdll -O -Wall -Ic:\Python\include -Ic:\Python\PC -c wrapper_module_5.c -o build\temp.win-amd64-2.7\Release\wrapper_module_5.o -std=c99
	wrapper_module_5.c:232:31: fatal error: numpy/arrayobject.h: No such file or directory
	compilation terminated.
	error: command 'c:\\Python\\MinGW\\bin\\gcc.exe' failed with exit status 1

	___________________________________________________________________________________________________________________________________
	____________________________________ sympy\external\tests\test_autowrap.py:test_ufuncify_numpy ____________________________________
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 167, in test_ufuncify_numpy
	    runtest_ufuncify('C', 'numpy')
	  File "d:\sympytest\sympy\sympy\external\tests\test_autowrap.py", line 93, in runtest_ufuncify
	    fabc = ufuncify([a, b, c], a*b + c, backend=backend)
	  File "sympy\core\cache.py", line 91, in wrapper
	    retval = cfunc(*args, **kwargs)
	  File "sympy\core\compatibility.py", line 857, in wrapper
	    return user_function(*args, **kwds)
	  File "sympy\utilities\autowrap.py", line 864, in ufuncify
	    return code_wrapper.wrap_code(routine, helpers=helps)
	  File "sympy\utilities\autowrap.py", line 151, in wrap_code
	    shutil.rmtree(workdir)
	  File "c:\Python\lib\shutil.py", line 252, in rmtree
	    onerror(os.remove, fullname, sys.exc_info())
	  File "c:\Python\lib\shutil.py", line 250, in rmtree
	    os.remove(fullname)
	WindowsError: [Error 5] Access is denied: 'c:\\users\\pecker\\appdata\\local\\temp\\tmpqfz1mc_sympy_compile\\wrapper_module_6.pyd'

	___________________________________________________________________________________________________________________________________
	____________________________ sympy\interactive\tests\test_ipythonprinting.py:test_print_builtin_option ____________________________
	  File "d:\sympytest\sympy\sympy\interactive\tests\test_ipythonprinting.py", line 70, in test_print_builtin_option
	    assert text in ("{pi: 3.14, n_i: 3}", u('{n\u1d62: 3, \u03c0: 3.14}'))
	AssertionError
	___________________________________________________________________________________________________________________________________
	______________________________ sympy\physics\vector\tests\test_printing.py:test_vector_pretty_print _______________________________
	  File "d:\sympytest\sympy\sympy\physics\vector\tests\test_printing.py", line 45, in test_vector_pretty_print
	    assert expected == pp.doprint(v)
	AssertionError
	___________________________________________________________________________________________________________________________________
	______________________________ sympy\physics\vector\tests\test_printing.py:test_dyadic_pretty_print _______________________________
	  File "d:\sympytest\sympy\sympy\physics\vector\tests\test_printing.py", line 138, in test_dyadic_pretty_print
	    assert expected == result
	AssertionError
	___________________________________________________________________________________________________________________________________
	___________________________________ sympy\plotting\tests\test_plot_implicit.py:test_matplotlib ____________________________________
	  File "d:\sympytest\sympy\sympy\plotting\tests\test_plot_implicit.py", line 63, in test_matplotlib
	    plot_and_save('test')
	  File "d:\sympytest\sympy\sympy\plotting\tests\test_plot_implicit.py", line 58, in plot_and_save
	    assert 'No labeled objects found' in str(w[0].message)
	AssertionError
	___________________________________________________________________________________________________________________________________
	____________________________________ sympy\vector\tests\test_printing.py:test_pretty_printing _____________________________________
	  File "d:\sympytest\sympy\sympy\vector\tests\test_printing.py", line 78, in test_pretty_printing
	    assert pretty(v[8]) == pretty_v_8
	AssertionError

	 tests finished: 5678 passed, 5 failed, 138 skipped, 321 expected to fail, 4 expected to fail but passed, 6 exceptions, 
	in 3051.11 seconds 
	DO *NOT* COMMIT!

--- a/sympy/utilities/autowrap.py
+++ b/sympy/utilities/autowrap.py
@@ -148,7 +148,11 @@ class CodeWrapper(object):
             CodeWrapper._module_counter += 1
             os.chdir(oldwork)
             if not self.filepath:
-                shutil.rmtree(workdir)
+                try:
+                    shutil.rmtree(workdir)
+                except OSError:
+                    # Could be some issues on Windows
+                    pass
 
         return self._get_wrapped_function(mod, routine.name)
 

